### ISSUE TITLE
Assigned to Field - Allow Queues 

### ISSUE THREAD

**Issue by etovar20** [SalesforceLabs/ActionPlansV4#38]
**Is your feature request related to a problem? Please describe.**
Unable to assign the tasks to a queue of users 

**Describe the solution you'd like**
allow the ability to assign the tasks to a queue of users rather then just individual

**Describe alternatives you've considered**
Considered creating a chatter free user and assigning to that but then unable to view those tasks 

**Additional context**
![image](https://user-images.githubusercontent.com/106348289/170564030-fdb5d91c-8d60-4578-90d3-54b13f79ea3c.png)

**Comment by dschach** [comment]
We don't currently support queues, but I will try to add that feature. I don't know if that will be possible on the ActionPlans page, but we can prevent changing the APTask User when the Task is reassigned to a queue. I don't know how to change the APTask object to handle a queue, but that's a possibility. Thank you.

**Comment by mhummell** [comment]
Prevent changing the APTask User when a Task is updated to a Queue would be **hugely helpful!**

I'm struggling with the same issue @etovar20 identified above. My business teams need to be able to select and assign Tasks from a shared Queue so this functionality is critical to us being able to fully implement Action Plans.

**Comment by mhummell** [comment]
@etovar20 hey just created a workaround that I wanted to share - I have a flow that reassigns the Action Plan created Tasks; if I have it clear the Action Plan ID field I'm able to reassign to a Queue without causing the Apex Trigger to fire.
![2022-06-06 12_19_09-Task_ IOC Daily Checklist - V4 and 9 more pages - Work - Microsoft​ Edge](https://user-images.githubusercontent.com/106839922/172202136-71c08e8f-da0e-4cff-9482-5f4838f926df.png)

**Comment by dschach** [comment]
Thank you all for your help. Yes, the Flow to update Task is the best way to go. I am working on a new release that will show the AP Task "Owner" as the Queue name if the related Task is owned by a Queue, and I'd like to prevent the user from editing that field unless the Task is owned by a User. It's complicated for other reasons, including the possibility that the parent record is owned by a Queue, in which case we cannot assign the APTasks to a Queue automatically - or at all. This may require a full rebuild of the APTask.User__c field eventually. For now, we will just make the following updates:

- If the parent record is owned by a Queue, assign the APTasks to the running user, and let customers make Flows to reassign the owner to a Flow. This avoids breaking Action Plans trying to assign APTasks to a Queue. 
- If the Task is assigned to a Queue, do not update the APTask User (we may clear the field in a revision) and display the Queue name read-only on the Action Plan Display and Edit pages.
- Prevent loops where we update the Task owner, which updates the APTask owner, which updates the Task owner... you get the idea.
- Avoid hard-coding anything so we can handle orgs that do not allow assigning Tasks to Queues, just as we are handling orgs that do.

I would like to find a way to have Queues involved in APTasks, but that would require replacing some fields, and while that's possible, it's best to save for the next release.

Thank you all for your help and patience. This is a revision of code from 2011, and ALL the logic was in the Visualforce controllers. While that is largely still the case, much of the action has been moved to trigger handlers and utility classes, and the hope is that eventually we can move almost all the code to handlers/utilities and can use very lightweight LWC for creating/editing these records. Maybe I should do a write-up of the complexities, since there are many ways to get to the New Action Plan screen.

To summarize: I would suggest making that Flow when a Task is created, especially if it is created via a Flow using the Action Plan invocable Apex Flow action. You're given the APTask IDs, and can possibly query for the related Tasks and then update those.

**Comment by mhummell** [comment]
@dschach sounds good, thank you for the response!

**Comment by etovar20** [comment]
@dschach @mhummell I thought of reassigning to the Queue and disassociating from the Action Plan, but then it will not update the action plan progress when the task(s) are completed, which is crucial to our business practice. I will stick to my workaround and wait for the next release.

Thanks for all of your help!

**Comment by dschach** [comment]
@mhummell  @etovar20 
A new version has been released in BETA. You should be able to put it into a sandbox. I'm not sure if it will install over your old version, but please give it a shot:
https://login.salesforce.com/packaging/installPackage.apexp?p0=04t5f000000rKeiAAE
